src/psbt.h

```diff
@@ -428,6 +433,12 @@ struct PartiallySignedTransaction
         OverrideStream<Stream> os(&s, s.GetType(), s.GetVersion() | SERIALIZE_TRANSACTION_NO_WITNESS);
         SerializeToVector(os, *tx);
 
+        // PSBT version
+        if (m_version != std::nullopt && m_version > 0) {
```

Comment:
I think it should be:

`if (m_version != std::nullopt && m_version > 0) {`
->
`if (m_version != std::nullopt && *m_version > 0) {`

at least your next commit makes me think so (https://github.com/bitcoin/bitcoin/pull/21283/commits/92ff609c1ef9d3cd848d08b4bbec1cdbf4ae4f0b#diff-2fb519d85c6ae99cd87c94e5b7cba43ff2ddd031b2c523dd2ffb12e991c16494R438)

Reply:
Done